coderd/coderdtest/authorize.go

```diff
@@ -245,6 +245,7 @@ func AGPLRoutes(a *AuthTester) (map[string]string, map[string]RouteCheck) {
 		// Endpoints that use the SQLQuery filter.
 		"GET:/api/v2/workspaces/":      {StatusCode: http.StatusOK, NoAuthorize: true},
 		"GET:/api/v2/workspaces/count": {StatusCode: http.StatusOK, NoAuthorize: true},
+		"GET:/api/v2/users/count":      {StatusCode: http.StatusOK, NoAuthorize: true},
```

Comment:
not doing an auth check here, combined with the fact that you can query for username or email substrings means that you can easily dump all usernames and emails by repeatedly hitting this endpoint and doing a tree search.

Reply:
To follow up, you can follow how we do this for workspaces.
https://github.com/coder/coder/blob/6f8ec23b17adb07f6005f214baab0fe71aaf8f3d/coderd/database/modelqueries.go#L113-L117

If you add `api.Authorize(r, rbac.ActionRead, rbac.ResourceUser)` it is technically ok as it checks if you can read all users. We should really do it proper though so we can mess with perms later and everything still works.

Add a new function called `GetAuthorizedUsersCount`